indico/modules/rb/models/reservation_occurrences.py

```diff
@@ -250,9 +251,17 @@ def can_reject(self, user, allow_admin=True):
         return self.reservation.can_reject(user, allow_admin=allow_admin)
 
     def can_cancel(self, user, allow_admin=True):
+        if user is None:
+            return False
         if not self.is_valid or self.end_dt < datetime.now():
             return False
-        return self.reservation.can_cancel(user, allow_admin=allow_admin)
+        resv = self.reservation
+        booked_or_owned_by_user = resv.is_owned_by(user) or resv.is_booked_for(user)
+        if booked_or_owned_by_user and self.start_dt >= datetime.now():
+            return True
+        if resv.is_rejected or resv.is_cancelled or resv.is_archived:
+            return False
+        return booked_or_owned_by_user or allow_admin and rb_is_admin(user)
```

Comment:
parentheses!

Reply:
Yeah, but the `and` operator has higher precedence anyway ;P

Reply:
Yes, it's well-defined but knowing the and/or precedence is something I consider "stuff you should not NEED to know" ;)